controllers/topics.js

```diff
@@ -16,7 +16,15 @@ exports.getArticleByArticleId = (req, res, next) => {
     const { article_id } = req.params
     selectArticleByArticleId(article_id)
     .then((article) => {
-        res.send(article.rows[0])
+        if (article.rows.length < 1) {
+            return Promise.reject({
+                status: 404,
+                msg: 'Article not found!'
+            })
+        } else {
+            article = article.rows[0]
+            res.send({ article })
+        }
```

Comment:
This promise rejection should be taking place in the model - otherwise you could just call `next` directly with your custom object.
Also, I think the parameter name `article` is misleading as it is an array at this point. Again, if this aspect is handled in the model, then you can have the whole promise resolve to an individual article object